Automattic/sensei
# Handle synchronization option change

Resolves #7235

## Proposed Changes
* Drop existing progress tables, re-create them and run migration, when sync has been enabled.
* Provide better messages for the migration status.

## Testing Instructions

0. Checkout the branch.
1. Build assets.
2. Enable experimental features UI: `add_filter( 'sensei_feature_flag_experimental_features_ui', '__return_true' );`
3. Go to Sensei LMS > Settings > Experimental Features.
4. If the sync checkbox is not checked, make sure the repository selection options are disabled.
5. When you check the sync and save, it is expected to see a few messages depending on the migration status (below the checkbox label):
  - Waiting for data migration to start.
  - Data migration is in progress. Please wait for it to switch repository.
  - Migration complete and data synchronization enabled.
  - Data migration failed. (and error messages below)
6. When you enabled the checkbox and switched to HPPS storage, make sure the sync checkbox is disabled.

## Pre-Merge Checklist
